metadata-ingestion/tests/unit/test_thrift.py

```diff
@@ -0,0 +1,33 @@
+import json
+
+from freezegun import freeze_time
+
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.source.thrift import ThriftSource
+from tests.test_helpers import mce_helpers
+
+FROZEN_TIME = "2020-04-14 07:00:00"
+
+
+@freeze_time(FROZEN_TIME)
+def test_thrift(
+    tmp_path,
+    pytestconfig,
+):
+
+    source = ThriftSource.create(
+        {"filename": "./examples/thrift_files/example.thrift"},
+        PipelineContext(run_id="test_run_id"),
+    )
+    mcp_objects = [wu.metadata.to_obj() for wu in source.get_workunits()]
+
+    with open(str(tmp_path / "thrift_mcps.json"), "w") as f:
```

Comment:
Instead of the golden files based testing, please write tests against the in-memory python objects directly. See [this](https://github.com/datahub-project/datahub/blob/master/smoke-test/tests/test_stateful_ingestion.py) for an example. Feel free to ask for more if needed.

Reply:
Hello, it sames that in the file you provided, the test is for checking whether pipline works correct, yes? In my test file, I don't use mysql pipline or any complex pipline, what should I check? I just read input file, ingest and compare with golden file.

Reply:
Does checkoutpoint(in line 98 of [this](https://github.com/datahub-project/datahub/blob/8d281fc013080e1f955107774f4c0ce9639aab81/smoke-test/tests/test_stateful_ingestion.py#L98)) contains information about whether the pipline work well and we use this to check whether pipline work well?

Reply:
In your test, you would be writing unit-tests directly against the key contents of the mcp_objects, such as the fieldPath, type and nativeDataType of the schemaFields produced. [This](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/tests/unit/test_kafka_source.py) is another example of such a test(we are generally trying to move away from golden file based tests, as they have a lot of irrelevant data that can change due to logic external to this connector). If you need help with a sample, I can help update this test itself, please let me know.

Reply:
Should I still read the same example thrift file, but only check whether some critical features(check whether second field's nativetype is i32 and whether second field's type is {"com.linkedin.schema.NumberType": {}}) are right instead of compare with golden file?

Reply:
Yes, that's correct. We basically don't want to use the golden files. It is perfectly okay to use as many thrift files as needed for the tests.